ospfd/ospf_vty.c

```diff
@@ -6656,8 +7245,16 @@ DEFUN (show_ip_ospf_instance_database_type_adv_router,
 		}
 
 		ret = show_ip_ospf_database_type_adv_router_common(
-			vty, ospf, idx ? 1 : 0, argc, argv, use_vrf);
+			vty, ospf, idx ? 1 : 0, argc, argv, use_vrf, json, uj);
+	}
+
+	if (uj) {
+		vty_out(vty, "%s\n",
+			json_object_to_json_string_ext(
+				json, JSON_C_TO_STRING_PRETTY));
```

Comment:
General tip: if the JSON output might be big I'd avoid using `JSON_C_TO_STRING_PRETTY` and I'd leave up to the user to use a JSON beautifier. Worst case is: with big pretty JSON output we use more memory and spend more CPU cycles sending to vtysh (we already had these problems in the past).

The beautified JSON consumes much more memory with indentation spaces and newlines.

Reply:
Agreed, Addressed.